Chat/AbstractSocket.cpp

```diff
@@ -34,10 +34,17 @@ bool AbstractSocket::Close()
     }
 
     int error_check = shutdown(socket_, SD_BOTH);
-    if(error_check == SOCKET_ERROR)
+    // 10057 - error socket not connected
+    //caused by shutdown+close on the other end of connection
+    if(error_check == SOCKET_ERROR && GetLastError() != 10057)
```

Comment:
Please, delete magic number.
You should better define that error code somewhere above.